### ISSUE TITLE
[Deprecation] Remove `rowSelectionMode` from Datagrid

### ISSUE THREAD

**Issue by hippee-lee** [vmware/clarity#2933]
## Summary

The `rowSelectionMode` Input should be removed from ClrDatagrid due to accessibility and maintenance issues. 

Removal of `@Input('clrDgRowSelection')` and the associated property on the `Selection` provider (rowSelectionMode).
* Why should it be removed from Clarity?
1. The clearly breaks accessibility in datagrid rows. 
2. Removes associated complexity in templates and rendering from the datagrid rendering cycle. 
* Does this change impact existing behaviors? If so how?
Yes, it removes the ability to click to select a row. 
* If this change introduces a new behavior, is this behavior accessible?
Yes.

## Use case

n/a

## Examples

n/a

## Workarounds

n/a
